### PR TITLE
Use memoryview when decoding HPACK headers

### PR DESCRIPTION
**Pull request by irvind**
It's almost the same pull request as https://github.com/Lukasa/hyper/pull/181 but for the correct repository. Instead of slicing I'm using memoryview to increase decoding perfomance.

### CODE REVIEW

File: hpack/hpack.py [hunk 1]
```diff
@@ -353,9 +354,10 @@ def decode(self, data, raw=False):
             current_index += consumed
 
         if raw:
-            return [(n, v) for n, v in headers]
+            return [(to_bytes(n), to_bytes(v)) for n, v in headers]
         else:
-            return [(n.decode('utf-8'), v.decode('utf-8')) for n, v in headers]
+            return [(to_bytes(n).decode('utf-8'), to_bytes(v).decode('utf-8'))
+                    for n, v in headers]
```
Comment (Lukasa) [inline hpack/hpack.py]: Hmm, this section is getting a bit complex. Can we try to hoist some of this functionality out into a function or a lambda that would let us save some lines? For example, a function that takes a name-value tuple and the `raw` parameter, and returns it coded appropriately.

**Comment by Lukasa** [comment]
Thanks for this @irvind! One small note. =)

**Comment by irvind** [comment]
@Lukasa Refined a little bit. Is it ok now?

**Comment by Lukasa** [comment]
Hurrah, looks good to me! Thanks @irvind!

**Pull request merged**